### DISCUSSION TITLE
Lecture 6 问题：Square MxM matrix

### DISCUSSION THREAD

**Discussion by luckyyang** [zkp-co-learning/zkp-co-learn#64] [Q&A（提问题在隔壁~）] [answered]
![2023-03-14 at 11 16](https://user-images.githubusercontent.com/1963646/224903241-cb16bc33-38a2-4354-af7c-85ed976285a2.png)
https://youtu.be/bTiNNw8lHpc?t=1778 时间: 29:38

左图中：
1. 这里的 $h_1, ..., h_M$ 是什么？每一项是 Square MxM matrix 中的每一行吗？
2. 下面乘积中的 $H_j$ 是被刻意构造出来的吗， $a_{ij}$ 是什么？

右图中：
1. 为什么要切分出那些 sets 呢？

**Comment by Howard-Hu** [comment]
这部分内容应该是对应这篇论文https://jbootle.github.io/Misc/pippenger.pdf 的第3节的内容，这部分内容我也还没有完全明白；蹲一个回答~

**Comment by crazysnut** [comment, marked as answer]
我来尝试回答下这个问题。

![image](https://user-images.githubusercontent.com/127089609/225804185-9dcdb644-3c65-4e1f-9db2-aaa3fe85e91d.png)
